### PR TITLE
Add Support for Smashing `SRP051449`

### PR DESCRIPTION
**Pull request by Miserlou**
## Issue Number

n/a

## Purpose/Implementation Notes
We can't rely on externally supplied `title` fields for smash column titles as they aren't guaranteed to be unique. Switches to use `accession_code`, which is. Also fixes an email bug case.

## Types of changes
- Bugfix (non-breaking change which fixes an issue)

## Functional tests

New test files and assertions added.

## Checklist

_Put an `x` in the boxes that apply._

- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: workers/data_refinery_workers/processors/smasher.py [hunk 1]
```diff
@@ -393,14 +393,14 @@ def _smash(job_context: Dict) -> Dict:
 
                     # Explicitly title this dataframe
                     try:
-                        data.columns = [computed_file.samples.all()[0].title]
+                        # Unfortuantely, we can't use this as `title` can cause a collision
+                        #data.columns = [computed_file.samples.all()[0].title]
+                        # So we use this, which also helps us support the case of missing SampleComputedFileAssociation
+                        data.columns = [computed_file.filename]
                     except ValueError:
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/smasher.py]: Is this except block still necessary? I think the only exception that could be raised here would be an AttributeError.

File: workers/data_refinery_workers/processors/smasher.py [hunk 2]
```diff
@@ -495,6 +497,10 @@ def _smash(job_context: Dict) -> Dict:
                         ks_test = rlang("ks.test")
 
                         set_seed(123)
+                        n = ncol(reso)
+                        m = 2
+                        if n < m:
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/smasher.py]: This could just be `if ncol(reso) < 2:` right?

File: workers/data_refinery_workers/processors/smasher.py [hunk 3]
```diff
@@ -495,6 +497,10 @@ def _smash(job_context: Dict) -> Dict:
...
+                            raise Exception("Found fewer columns that required for QN combinatorial - bad smash?")
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/smasher.py]: `than required`*

File: workers/data_refinery_workers/processors/utils.py [hunk 4]
```diff
@@ -152,7 +152,7 @@ def end_job(job_context: Dict, abort=False):
         # S3-sync Computed Files
         for computed_file in job_context.get('computed_files', []):
             # Ensure even distribution across S3 servers
-            nonce = ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(8))
+            nonce = ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(24))
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/utils.py]: Just curious why this change?

File: workers/data_refinery_workers/processors/test_smasher.py [hunk 5]
```diff
@@ -610,6 +610,103 @@ def test_no_smash_dupe(self):
         for column in final_context['original_merged'].columns:
             self.assertTrue('_x' not in column)
 
+    @tag("smasher")
+    def test_no_smash_dupe_two(self):
```
Comment (kurtwheeler) [inline workers/data_refinery_workers/processors/test_smasher.py]: I'm just wondering, what's this testing that isn't already tested? I'm not quite sure what the test function's name means.

**Comment by jaclyn-taroni** [comment]
👍

File: workers/data_refinery_workers/processors/test_smasher.py (hunk 5 above)
Comment (Miserlou) [reply workers/data_refinery_workers/processors/test_smasher.py]: The test reflects an actual situation encountered where the tiles are the same, it also uses a QN target and has qn_normalize `true` on the dset

File: workers/data_refinery_workers/processors/utils.py (hunk 4 above)
Comment (Miserlou) [reply workers/data_refinery_workers/processors/utils.py]: 3 million titles.. just wanted to expand the space

File: workers/data_refinery_workers/processors/smasher.py (hunk 2 above)
Comment (Miserlou) [reply workers/data_refinery_workers/processors/smasher.py]: Had to add a [0] to this anyway

File: workers/data_refinery_workers/processors/smasher.py (hunk 1 above)
Comment (Miserlou) [reply workers/data_refinery_workers/processors/smasher.py]: Changed to base exception, which I think is necessary for two-channel samples

File: workers/data_refinery_workers/processors/test_smasher.py (hunk 5 above)
Comment (kurtwheeler) [reply workers/data_refinery_workers/processors/test_smasher.py]: Ah okay. I don't think there's anything in the code which explains this. The code looks like it's operating on two different files. If you could add a comment or docstring that would be helpful.
